### PR TITLE
[#1860] feat(CLI/Coordinator/Dashboard): Add Reg time to AppInfo and Application

### PR DESCRIPTION
**Pull request by maobaolong**
### What changes were proposed in this pull request?

Add Reg time to AppInfo and Application

### Why are the changes needed?

Fix: #1860

### Does this PR introduce _any_ user-facing change?

`bin/uniffle  client-cli -app -host localhost -port 19998` and dashboard app page will display registrationTime of each application.

### How was this patch tested?

```Console
bin/uniffle  client-cli -app -host localhost -port 19998
host:localhost,port:19998
+--------------------------------------------------------------------------------------------------------------+
|                                             Uniffle Applications                                             |
+---------------------------------------+------+---------------------+---------------------+-------------------+
|             ApplicationId             | User |    Registration Time    | Last HeartBeatTime  | RemoteStoragePath |
+---------------------------------------+------+---------------------+---------------------+-------------------+
| app-20240703231912-0003_1720019950472 | user | 2024-07-03 23:19:40 | 2024-07-03 23:21:05 |       null        |
| app-20240703232127-0004_1720020086199 | user | 2024-07-03 23:21:51 | 2024-07-03 23:22:16 |       null        |
+---------------------------------------+------+---------------------+---------------------+-------------------+
```

image

### CODE REVIEW

**Comment by rickyma** [comment]
This is good. PTAL. @xianjingfeng @yl09099

File: common/src/main/java/org/apache/uniffle/common/Application.java [hunk 1]
```diff
@@ -36,12 +37,14 @@ public Application(Builder builder) {
...
     this.lastHeartBeatTime = builder.lastHeartBeatTime;
     this.remoteStoragePath = builder.remoteStoragePath;
+    this.registerTime = builder.registerTime;
   }
 
   public static class Builder {
     private String applicationId;
     private String user;
     private String lastHeartBeatTime;
+    private String registerTime;
```
[the remark this answers is not in the document]
Comment (maobaolong) [reply common/src/main/java/org/apache/uniffle/common/Application.java]: Done

**Comment by maobaolong** [comment]
@rickyma Thanks for your review, addressed your comments, PTAL

**Review by xianjingfeng — APPROVED** [review]
LGTM

**Comment by yl09099** [comment]
LGTM

**Pull request merged**